api/v1alpha1/application.go

```diff
@@ -3,35 +3,68 @@ package v1alpha1
 import (
 	"golang.org/x/exp/constraints"
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-//+kubebuilder:object:root=true
+// +kubebuilder:object:root=true
 type ApplicationList struct {
 	metav1.TypeMeta `json:",inline"`
 	metav1.ListMeta `json:"metadata,omitempty"`
 
 	Items []Application `json:"items"`
 }
 
-//+kubebuilder:object:root=true
-//+kubebuilder:resource:shortName="app"
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:shortName="app"
 type Application struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
 	Spec ApplicationSpec `json:"spec,omitempty"`
 }
 
-//+kubebuilder:object:generate=true
+// +kubebuilder:object:generate=true
+type ResourceRequirements struct {
+	//+kubebuilder:validation:Required
+	Limits LimitResourceList `json:"limits"`
+	//+kubebuilder:validation:Required
+	Requests RequestsResourceList `json:"requests"`
+}
+
+// +kubebuilder:object:generate=true
+type LimitResourceList struct {
+	//+kubebuilder:validation:Optional
+	CpuLimit resource.Quantity `json:"cpu,omitempty"`
+	//+kubebuilder:validation:Required
+	MemoryLimit resource.Quantity `json:"memory"`
+	//+kubebuilder:validation:Optional
+	StorageLimit resource.Quantity `json:"storage,omitempty"`
+	//+kubebuilder:validation:Optional
+	EphemeralStorageLimit resource.Quantity `json:"ephemeralStorage,omitempty"`
```

Comment:
Should we support `EphemeralStorageLimit` and `StorageRequest`?

Reply:
Is it currently unsupported?

Reply:
Not documented, so not explicitly supported at least

Reply:
if you prefer i could remove these. thought if they're not currently unsupported i dont see why we should, in case they might be useful for some case